lib/lifecycles/bump.js

```diff
@@ -20,16 +20,45 @@ async function Bump (args, version) {
   configsToUpdate = {}
 
   if (args.skip.bump) return version
+
+  if (args.releaseAs && !(['major', 'minor', 'patch'].includes(args.releaseAs.toLowerCase()) || semver.valid(args.releaseAs))) {
+    throw Error("releaseAs must be one of 'major', 'minor' or 'patch', or a valid semvar version.")
+  }
+
   let newVersion = version
   await runLifecycleScript(args, 'prerelease')
   const stdout = await runLifecycleScript(args, 'prebump')
-  if (stdout && stdout.trim().length) args.releaseAs = stdout.trim()
-  const release = await bumpVersion(args.releaseAs, version, args)
+  if (stdout?.trim().length) {
+    const prebumpString = stdout.trim()
+    if (semver.valid(prebumpString)) args.releaseAs = prebumpString
+  }
   if (!args.firstRelease) {
-    const releaseType = getReleaseType(args.prerelease, release.releaseType, version)
-    const releaseTypeAsVersion = releaseType === 'pre' + release.releaseType ? semver.valid(release.releaseType + '-' + args.prerelease + '.0') : semver.valid(releaseType)
+    if (semver.valid(args.releaseAs)) {
+      const releaseAs = new semver.SemVer(args.releaseAs)
+      if (isString(args.prerelease)) {
+        if (releaseAs.prerelease.length) {
+          if (releaseAs.prerelease.slice(0, -1).join('.') !== args.prerelease) {
+            throw Error('releaseAs and prerelease have conflicting prerelease identifiers')
+          }
+          newVersion = releaseAs.version
+        } else {
+          newVersion = `${releaseAs.major}.${releaseAs.minor}.${releaseAs.patch}-${args.prerelease}.0`
+        }
+        // Check if the previous version is the same version and prerelease, and increment if so
+        if (['prerelease', null].includes(semver.diff(version, newVersion)) && semver.lte(newVersion, version)) {
+          newVersion = semver.inc(version, 'prerelease', args.prerelease)
+        }
+        // Append any build info from releaseAs
+        newVersion = [newVersion, releaseAs.build.join('.')].filter(Boolean).join('+')
+      } else {
+        newVersion = [releaseAs.version, releaseAs.build.join('.')].filter(Boolean).join('+')
+      }
+    } else {
```

Comment:
Optional: This series of if statements is a bit hard to understand, as it's hard to spot which else belongs to which condition. Some ways we could address this:

* Maybe it would be better as a number of `if / else if` where the full condition was repeated? 
* Or it might be improved with comments at the else block explaining the condition
* Or perhaps some sections should be pulled out into functions for readability?

Reply:
I'll do a readability pass and see what I can do, I agree that this wasnt the clearest.

Reply:
I've replaced it with an elif chain that I think is a small improvement, though the entire lifecycle could probably be improved for readability further through a restructure in future.